CLI: add documentation for Cody CLI (#502)
sourcegraph/docs: docs/cody/clients/install-cli.mdx

# Install Cody CLI

> <p className="subtitle">Learn how to install the <code>cody</code> command-line tool.</p>

Reviewer: ```suggestion
<p className="subtitle">Learn how to install the <code>cody</code> command-line tool and using the <code>cody chat</code> subcommand.</p>
```

<Callout type="note" title="Experimental">

> Cody CLI support is in the experimental stage.

Reviewer: We should also mention what pricing tiers will it support in the callout.

Author: I added

> Cody CLI is available to Free, Pro, and Enterprise customers.

The `--context-repo` section highlights that this specific feature is limited to Enterprise users (just like in the IDE)

</Callout>

> The Cody extension by Sourcegraph enhances your coding experience in your IDE by providing intelligent code suggestions, context-aware completions, and advanced code analysis. This guide will walk you through the steps to install and set up Cody within your JetBrains environment.

Reviewer 2: This text feels wrong - I think we should talk about the CLI here.

Author: good catch, I copy-pasted the JB page. Wrote instead

> Cody CLI is the same technology that powers the Cody IDE plugins but available from the command-line.
> Use Cody CLI for ad-hoc exploration in your terminal or as part of scripts to automate your workflows.

<LinkCards>
   <LinkCard href="https://www.npmjs.com/package/@sourcegraph/cody" imgSrc="https://storage.googleapis.com/sourcegraph-assets/docs/images/cody/command-line-blue.svg" imgAlt="Cody CLI" title="Cody CLI" description="Install Cody CLI from npm" />
</LinkCards>

## Prerequisites

> - You have Node.js v20 or newer installed

Reviewer: ```suggestion
- You have Node.js `v20` or newer installed
```

Author: Done.

cody auth login # stores the access token securely
```
You don't need to export `SRC_ENDPOINT` or `SRC_ACCESS_TOKEN` after running `cody auth login`.
Subsequent command invocation will use the stored access token.
  </Tab>
</Tabs>

Confirm that the authentication was successful by running the following command:

```shell
cody auth whoami
# ✔ Authenticated as USERNAME on ENDPOINT
```

## Authenticate with only environment variables

Skip this step if you have already authenticated with the `cody auth login` command.

If you prefer not to let Cody CLI store your access token, you can also pass the endpoint URL and access token through the environment variables `SRC_ENDPOINT` and `SRC_ACCESS_TOKEN`.

<Tabs>
  <Tab title="macOS/Linux">
```shell
export SRC_ENDPOINT=ENDPOINT
export SRC_ACCESS_TOKEN=ACCESS_TOKEN
```
  </Tab>
  <Tab title="Windows">
```powershell
$env:SRC_ENDPOINT = "ENDPOINT"
$env:SRC_ACCESS_TOKEN = "ACCESS_TOKEN"
```
  </Tab>
</Tabs>

It's recommended to store these access tokens in a secure location. 
For example, you can store them with a password manager like [1Password](https://1password.com/) or [Bitwarden](https://bitwarden.com/).

> It may be tempting to export these variables in your shell startup script, but this is not recommended because it will expose your access token to all commands you run from the terminal.

Reviewer: ```suggestion
It is not recommended to export these variables in your shell startup script because it will expose your access token to all commands you run from the terminal. Instead, consider sourcing these environment variables on-demand when you need to authenticate with the Cody CLI.
```

Author: Much clearer, thank you.

Instead, consider sourcing these environment variables on-demand when you need to authenticate with the Cody CLI.

## Sign out with cody auth logout

To sign out of the Cody CLI, run the following command:

```shell
cody auth logout
```

Running this command wil remove the access token from the secure storage of your operating system.

Run `cody auth whoami` to confirm that the Cody CLI is no longer authenticated.

> # Chat with Cody CLI

Reviewer: ```suggestion
## Chat with Cody CLI
```
Only one H1 per page. Rest should be H2s or H3s. :)

Author: Fair enough. I've written a lot of docs with Docusaurus in the past where H2/H3 appear in the sidebar, not the H1. It's useful to be able to group sidebar items into sections like installation, chat, ... The sidebar looks like this now

![CleanShot 2024-07-18 at 14 27 41@2x](https://github.com/user-attachments/assets/d3ae571d-1be3-4370-bb12-d38917dc8b80)

> <p className="subtitle">Learn how to use the <code>cody chat</code> subcommand.</p>

Reviewer: ```suggestion
```
Likewise, only one entry for subtitle that summarizes the gist of the docs. So moving this to the top. 🙌

Author: Removed!

Once you've authenticated the Cody CLI, you can start using it to chat with Cody.

To start a new chat, run the following command:

```shell
cody chat -m 'Explain React hooks'
```

The following commands are equivalent to the above:

```shell
# use --message instead of -m
cody chat --message 'Explain React hooks'

# space separated arguments
cody chat Explain React hooks
```

## cody chat --context-file

Use `--context-file` to provide context from local files

```shell
cody chat --context-file src/controller.ts -m 'Are there code smells in this file?'
```

## cody chat --context-repo

> 

Reviewer: ```suggestion
<Callout type="info">This configuration is only supported on Cody Enterprise.</Callout>
```

Author: Much nicer 😍 

![CleanShot 2024-07-18 at 14 30 41@2x](https://github.com/user-attachments/assets/9e599a74-3752-4b6a-8dd6-bb06ac3b1635)

Use `--context-repo` to provide context from remote repositories.

> This feature is only available in Cody Enterprise.

Reviewer: ```suggestion

```

Author: Done.
